### DISCUSSION TITLE
Promt after defer

### DISCUSSION THREAD

**Discussion by JMCSFR** [PSAppDeployToolkit/PSAppDeployToolkit#744] [Q&A] [answered]
Hello,

How can I force Installation Welcome screen after x defers?

The issue is that an user defer x time the installation and after x+1 the installation starts imidiately without an information. The idea is to show to user Installation Welcome screen with greyedout defer button but with the information that after an user clicks to Continue the installation starts.

Thank you.
Jakub

**Comment by mjr4077au** [comment]
The toolkit should automatically do this once the deferral count has ran out.

**Reply by Pollewops** [reply]
This is not happening. When the deferrals are ended, the next time you will not see a popup anymore.
I created myself a workaround for it in the script.

**Reply by mjr4077au** [reply]
Do you mind presenting your workaround here for review?

**Reply by Pollewops** [reply]
This is what i used. Now i do not use it anymore since we stopped using defers due to moving to Intune. But this works fine with defers.


$deferTimesRemaining = (Get-DeferHistory).DeferTimesRemaining
    If ($DoInstall) {
        If ($deferTimesRemaining -eq 0) {
$MessageBox = "You are out of your defer counts now.
 
The following application is about to be installed:
 
$appName $appVersion
 
Please save your work, close the programs, and then click Continue.
 
"
                                $Result = Show-InstallationPrompt -Message $MessageBox -ButtonMiddleText 'Continue' -MessageAlignment 'Center'
                  } Else {
                      Show-InstallationWelcome -AllowDefer -DeferTimes $DeferCount -CheckDiskSpace -PersistPrompt -MinimizeWindows $false
        }
        Show-InstallationProgress -StatusMessage "Installation in progress..."

**Reply by mjr4077au** [reply]
Honestly the behaviour of the toolkit is probably by design. I see you're not using any blocking apps, therefore if nothing's blocking, the toolkit simply commences the install.

**Reply by DanGough** [reply, marked as answer]
The way it is right now, and how I believe it was designed, is to only show the prompt if there is either an option to defer or an application to close.

Your message is saying "Please save your work, close the programs, and then click Continue". If you add the programs that need to close via -CloseApps, then the prompt should display if they have the apps open. If they don't, there is no need to tell the users to save their work, so the installation can just proceed?

**Comment by Pollewops** [comment]
Correct assumption. So not required anymore unless you want to display a message since a reboot is required.Met een glimlach,JohanOp 6 nov 2024 om 16:30 heeft Dan Gough ***@***.***> het volgende geschreven:﻿
The way it is right now, and how I believe it was designed, is to only show the prompt if there is either an option to defer or an application to close.
Your message is saying "Please save your work, close the programs, and then click Continue". If you add the programs that need to close via -CloseApps, then the prompt should display if they have the apps open. If they don't, there is no need to tell the users to save their work, so the installation can just proceed?
